symphony-bdk-core/src/test/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV1Test.java

```diff
@@ -0,0 +1,282 @@
+        when(datafeedApi.v4DatafeedIdReadGet("test-id", "1234", "1234", null))
+                .thenThrow(new ApiException(400, "test_client_error"));
+        doNothing().when(authSession).refresh();
+
+        assertThrows(ApiException.class, this.datafeedService::start);
+        verify(datafeedApi, times(3)).v4DatafeedCreatePost("1234", "1234");
+        verify(datafeedApi, times(2)).v4DatafeedIdReadGet("test-id", "1234", "1234", null);
+    }
+
+    @Test
+    void startTestFailedAuthCreate() throws ApiException, AuthUnauthorizedException {
+        when(datafeedApi.v4DatafeedCreatePost("1234", "1234")).thenThrow(new ApiException(401, "test_unauthorized"));
+        doThrow(AuthUnauthorizedException.class).when(authSession).refresh();
+
+        assertThrows(AuthUnauthorizedException.class, this.datafeedService::start);
+        verify(datafeedApi, times(1)).v4DatafeedCreatePost("1234", "1234");
+    }
+
+    @Test
+    void startTestClientErrorCreate() throws ApiException, AuthUnauthorizedException {
+        when(datafeedApi.v4DatafeedCreatePost("1234", "1234")).thenThrow(new ApiException(400, "test_client_error"));
+
+        assertThrows(ApiException.class, this.datafeedService::start);
+        verify(datafeedApi, times(1)).v4DatafeedCreatePost("1234", "1234");
+    }
+
+    @Test
+    void startTestFailedAuthRead() throws ApiException, AuthUnauthorizedException {
+        when(datafeedApi.v4DatafeedCreatePost("1234", "1234")).thenReturn(new Datafeed().id("test-id"));
+        when(datafeedApi.v4DatafeedIdReadGet("test-id", "1234", "1234", null))
+                .thenThrow(new ApiException(401, "test_client_error"));
+        doThrow(AuthUnauthorizedException.class).when(authSession).refresh();
+
+        assertThrows(AuthUnauthorizedException.class, this.datafeedService::start);
+        verify(datafeedApi, times(1)).v4DatafeedCreatePost("1234", "1234");
+        verify(datafeedApi, times(1)).v4DatafeedIdReadGet("test-id", "1234", "1234", null);
+    }
+
+    @Test
+    void startServiceAlreadyStarted() throws ApiException, AuthUnauthorizedException {
+        AtomicInteger signal = new AtomicInteger(0);
+        List<V4Event> events = new ArrayList<>();
+        V4Event event = new V4Event().type(DatafeedEventConstant.MESSAGESENT);
+        events.add(event);
+        when(datafeedApi.v4DatafeedCreatePost("1234", "1234")).thenReturn(new Datafeed().id("test-id"));
+        when(datafeedApi.v4DatafeedIdReadGet("test-id", "1234", "1234", null))
+                .thenReturn(events);
+
+        this.datafeedService.unsubscribe(this.listener);
+        this.datafeedService.subscribe(new DatafeedEventListener() {
+            @Override
+            public void onMessageSent(V4Event event) {
+                try {
+                    datafeedService.start();
+                } catch (AuthUnauthorizedException | ApiException e) {
+                    e.printStackTrace();
```

Comment:
(it might seem less important in tests but if we have the habit of not using it every where it will be easier)
and we also don't want to pollute test outputs with logs and stacktraces